src/theme/components/socialmedialinks.ts

```diff
@@ -0,0 +1,185 @@
+// maybe set the border to always be on, but invisible if prop is set to "none"?
+
+import { StyleFunctionProps } from "@chakra-ui/theme-tools";
+
+function getRadius(size) {
+  let radiusSize = "119.125rem"; // 18px @todo Seems to be non standard. Should it be 16px / 1rem / 's'?
+  switch (size) {
+    case "medium":
+      radiusSize = "m"; // 24px
+      break;
+    case "large":
+      radiusSize = "l"; // 32px
+      break;
+  }
+  return {
+    radiusSize,
+  };
+}
+
+function getColor(color) {
+  let ltColor = "ui.typography.heading",
+    dkColor = "dark.ui.typography.heading",
+    ltBgColor,
+    dkBgColor;
+
+  switch (color) {
+    case "link":
+      ltColor = "ui.link.primary";
+      dkColor = "dark.ui.link.primary";
+      break;
+    case "textInverse":
+      ltColor = "ui.typography.inverse.heading";
+      dkColor = "dark.ui.typography.inverse.heading";
+      ltBgColor = "ui.typography.heading";
+      dkBgColor = "dark.ui.typography.heading";
+  }
+
+  return {
+    ltColor,
+    dkColor,
+    ltBgColor,
+    dkBgColor,
+  };
+}
+
+const SocialMediaLinks = {
+  // The parts of the component
+  parts: ["li", "a", "svg", "span"],
+  // The base styles for each part
+  baseStyle: {
+    // ul styles should not specifically identified as ul:
+    display: "flex",
+    flexDirection: "row",
+    alignItems: "flex-start",
+    gap: "s",
+    li: {
+      marginInlineEnd: "0",
+      marginTop: "0",
+      _notFirst: {
+        marginTop: "0",
+      },
+    },
+    a: {
+      display: "flex",
+      alignItems: "center",
+      gap: "xs",
+      alignSelf: "stretch",
+      fontStyle: "normal",
+      fontSize: "desktop.body.body2",
+      textDecoration: "none",
+      color: "ui.typography.heading",
+      _dark: {
+        color: "dark.ui.typography.heading",
+      },
+    },
+    svg: {
+      fill: "ui.typography.heading",
+      _dark: {
+        fill: "dark.ui.typography.heading",
+      },
+    },
+  },
+  // The variants are bordered
+  // @todo The borders add 2px to total width of each <li />, e.g. small w/ borders = 36px. But the figma has it at 34px.
```

Comment:
@bigfishdesign13 can better answer this.

Reply:
So here is the problem. In order to make the border appear inside the box (using `boxSizing: "border-box"`) I need to set the height and width of the box as absolute values. Currently they are set to `fit-content`. But I cannot know the width of the box if it uses labels.

Reply:
Sorry, I missed this.  This is an area where Figma and real-life HTML are not fully in sink.  That being the case, leave it as is.  The two pixel descrepancy between the Figma component and the React component is acceptable.